plugin/src/main/groovy/com/novoda/gradle/command/Input.groovy

```diff
@@ -1,80 +1,27 @@
 package com.novoda.gradle.command
 
+import org.gradle.api.Action
 import org.gradle.api.tasks.TaskAction
 
 class Input extends AdbTask {
 
-    Closure script
+    InputSpec script = new InputSpec()
 
-    void text(String value) {
-        input('text', "$value")
-    }
-
-    void tap(int x, int y) {
-        input('touchscreen', 'tap', x, y)
-    }
-
-    void swipe(int startX, int startY, int endX, int endY) {
-        input('touchscreen', 'swipe', startX, startY, endX, endY)
-    }
-
-    void key(int code) {
-        input('keyevent', code)
-    }
-    
-    void home() {
-        key 3
-    }
-    
-    void back() {
-        key 4
-    }
-
-    void up() {
-        key 19
-    }
-
-    void down() {
-        key 20
-    }
-    
-    void left() {
-        key 21
-    }
-
-    void right() {
-        key 22
-    }
-
-    void power() {
-        key 26
-    }
-    
-    void clear() {
-        key 28
-    }
-
-    void tab() {
-        key 61
-    }
-    
-    void enter() {
-        key 66
-    }
-
-    void unlock() {
-        key 82
-    }
-
-    private input(... values) {
-        def command = ["shell", "input"]
-        command.addAll(values)
-        runCommand(command)
+    @Deprecated
```

Comment:
This method has been just added and already deprecated: what's the point of it? 😕 
If you want to introduce this change without breaking any public API you should keep all the public methods deleted, mark them as `@Deprecated` and delegate to the `InputSpec` instead. Other choice will be to annotate the `script` field [as `@Delegate`](http://mrhaki.blogspot.de/2009/08/groovy-goodness-delegate-to-simplify.html) and let Groovy do its magic, but then the users won't have any notion of deprecation.